### ISSUE TITLE
Fullscreen when using multiple displays

### ISSUE THREAD

**Issue by fbongcam** [iina/iina#2167]
**System and IINA version:**

- macOS 10.14.2
- IINA 1.0.0

**Expected behavior:**
Entering fullscreen mode should play video in fullscreen on specified monitor when using multiple displays. There should be an option to choose which monitor to use for fullscreen. For example if I want to watch a video in fullscreen on **Monitor 1** and have **Monitor 2** available to be able to continue working.

**Actual behavior:**
Entering fullscreen mode when using multiple displays makes all displays black and unavailable to use for multitasking.

<details>
<summary><b>Crash report:</b></summary>

```
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

- [ ] MPV does not have this problem.

**How often does this happen?**

**Comment by anohren** [comment]
> Entering fullscreen mode when using multiple displays makes all displays black and unavailable to use for multitasking.

That sounds unusable. 

I've never had any problem dragging the fullscreen window to whichever monitor I want, using mission control. Do you have legacy fullscreen activated in settings?

**Comment by fbongcam** [comment]
Activating legacy fullscreen does work the way I'd like! Why wouldn't this just be the preferred/default fullscreen mode?

An option to specify which monitor to use as default fullscreen viewer would still be useful in my opinion.

**Comment by ErikMinekus** [comment]
Go to System Preferences > Mission Control and check "Displays have separate Spaces". Then you can fullscreen an application on one monitor and continue working on the other monitors.

> Why wouldn't this just be the preferred/default fullscreen mode?

Because it's legacy. This "new" fullscreen mode was added to macOS in 2011, and most applications have switched to it by now.

**Comment by rr0ss0rr** [comment]
I agree that it would be nice to always default to the external monitor without the need for dragging.  I also found that legacy FS changes from the external to internal monitor when proceeding to the next video in the playlist (only sometimes)

**Comment by fbongcam** [comment]
> Go to System Preferences > Mission Control and check "Displays have separate Spaces". Then you can fullscreen an application on one monitor and continue working on the other monitors.


Unfortunately this is not an option for me as it messes up my workflow in other applications.


> Ok, I was under the impression that disabling legacy fullscreen would fix the problem.


For me it's the other way around. Having legacy fullscreen checked fixes the problem for me.


> Now that I look in settings, there is an option to blank out other monitors when entering fullscreen. You should just have to disable that and there should be no need for legacy fullscreen.


I don't have this checked. When I have legacy fullscreen enabled this option gives me the same behaviour as non legacy and doesn't do anything when I'm using non legacy.

VLC doesn't have this bahaviour for example.

On a side note, having an option to chose UI language would make troubleshooting much easier as you wouldn't have to translate the meaning of every option.

**Comment by anohren** [comment]
> _On a side note, having an option to chose UI language would make troubleshooting much easier as you wouldn't have to translate the meaning of every option._

Seems like someone had the same idea: https://github.com/iina/iina/pull/2091 
I added this argument to that thread.

**Comment by 2600box** [comment]
Enabling Legacy Fullscreen does the job of removing the black monitors. Looking forward to further options on specifying which monitor is used for full-screen á la VLC.

**Comment by babalooza** [comment]
Is there a way to have both of the following features:
• have IINA always opening a movie on the external display (in fullscreen)
• quit IINA with no open windows

Currently the dock setting to have IINA running on the external display, specifically to open movies in fullscreen mode on the external display, is respected only with IINA's option to use the alternative or legacy fullscreen mode. But that option also prevents IINA to quit with closing its last window. I love love love IINA, but this behavior drives me nuts.

IINA 1.1.2 on macOS 10.15.7

**Comment by PaulSlocum** [comment]
I'm having this problem now -- IINA always takes over both monitors when going full screen and one screen is just black, which makes IINA useless.  Restarting the computer doesn't help.  I'm using the latest version of IINA (1.3.5) and VLC doesn't have this problem.   I can't find setting for "legacy full screen mode" in Sequoia 15.1
